src/garage/tf/policies/gaussian_lstm_policy_with_model.py

```diff
@@ -0,0 +1,295 @@
+            [
+                self.model.networks['default'].sample,
+                self.model.networks['default'].step_mean,
+                self.model.networks['default'].step_log_std,
+                self.model.networks['default'].step_hidden,
+                self.model.networks['default'].step_cell
+            ],
+            feed_list=[step_input_var, step_hidden_var, step_cell_var])
+
+        self.prev_actions = None
+        self.prev_hiddens = None
+        self.prev_cells = None
+
+    @property
+    def vectorized(self):
+        """Vectorized or not."""
+        return True
+
+    def dist_info_sym(self, obs_var, state_info_vars, name=None):
+        """
+        Symbolic graph of the distribution.
+
+        Args:
+            obs_var (tf.Tensor): Tensor input for symbolic graph.
+            state_info_vars (dict): Extra state information, e.g.
+                previous action.
+            name (str): Name for symbolic graph.
+
+        """
+        if self._state_include_action:
+            prev_action_var = state_info_vars['prev_action']
+            prev_action_var = tf.cast(prev_action_var, tf.float32)
+            all_input_var = tf.concat(
+                axis=2, values=[obs_var, prev_action_var])
+        else:
+            all_input_var = obs_var
+
+        with tf.variable_scope(self._variable_scope):
+            _, mean_var, _, log_std_var, _, _, _, _, _, _ = self.model.build(
+                all_input_var,
+                self.model.networks['default'].step_input,
+                self.model.networks['default'].step_hidden_input,
+                self.model.networks['default'].step_cell_input,
+                name=name)
+
+        return dict(mean=mean_var, log_std=log_std_var)
+
+    def reset(self, dones=None):
+        """
+        Reset the policy.
+
+        Args:
+            dones (numpy.ndarray): Bool that indicates terminal state(s).
+
+        """
+        if not dones:
+            dones = np.array([True])
+        if self.prev_actions is None or len(dones) != len(self.prev_actions):
+            self.prev_actions = np.zeros((len(dones),
+                                          self.action_space.flat_dim))
+            self.prev_hiddens = np.zeros((len(dones), self._hidden_dim))
+            self.prev_cells = np.zeros((len(dones), self._hidden_dim))
+
+        self.prev_actions[dones] = 0.
+        self.prev_hiddens[dones] = self.model.networks[
+            'default'].init_hidden.eval()
+        self.prev_cells[dones] = self.model.networks['default'].init_cell.eval(
+        )
+
+    @overrides
+    def get_action(self, observation):
+        """
+        Get single action from this policy for the input observation.
+
+        Args:
+            observation (numpy.ndarray): Observation from environment.
+
+        Returns:
+            action (numpy.ndarray): Predicted action.
+            agent_info (dict[numpy.ndarray]): Mean and log std of the
```

Comment:
be explicit
https://github.com/rlworkgroup/garage/blob/master/src/garage/tf/algos/batch_polopt.py#L85